Add react-native-svg props support on StackedBarChart component

LRNZ09 opened the pull request:
Most of the changes are made by ESlint.
Breaking changes are on line 163 and 218, since the prop `data` now is like the one that other charts need, so we can also support `svg` property in the future.

Before:
```javascript
[{ keyA: 123, keyB: 456, keyC: 789 }]
```
After:
```javascript
[{
  onPress: () => {},
  // svg: { ... }
  value: { keyA: 123, keyB: 456, keyC: 789 },
}]
```

JesperLekland commented:
Hi @LRNZ09 and a big thank you for your contribution 🥇 
 
I think you are on the right track but I don't feel like the extra properties belong on each `data` entry (what if I want each "part" of a bar to support `onPress`?). Rather I think that a more intuitive (and library aligned) way of doing would be to have each key entry support an object in addition to the integer along with a `valueAccessor` (defaults to `entry => entry`. This way we don't introduce any breaking changes either.

Before:
```javascript
[{ keyA: 123, keyB: 456, keyC: 789 }]
```

After: 
```javascript
[{keyA: { value: 123, svg: {...}, onPress: () => {}}}, ...]
...
valueAccessor={entry => entry.value}
```

Any thoughts on the above?

LRNZ09 commented:
Okay sounds good to me!
But I don't understand if `valueAccessor` needs to be passed as a prop from outside as a custom function when using the "new format" with the `onPress` callback.
Is there any of the other charts that have the same behavior?

JesperLekland commented:
`valueAccessor` should be the same concept as the other chart's `xAccessor`/`yAccessor`.

LRNZ09 commented on src/stacked-bar-chart.js:
```diff
@@ -179,28 +194,28 @@ class BarChart extends PureComponent {
                 <View style={{ flex: 1 }} onLayout={ event => this._onLayout(event) }>
-                    {
-                        height > 0 && width > 0 &&
+                    {height > 0 &&
```
Sorry, this is caused by ESLint

JesperLekland replied:
hmm.. then it should behave the same on my computer. Running `eslint --fix` on dev results on no changes. I think you might have some editor settings that are interfering. Please don't commit those changes (i.e revert them)

LRNZ09 commented on storybook/stories/index.js:
```diff
@@ -45,53 +46,54 @@ import PartialAreaChart from './partial-chart/area-chart'
+    .add('Grid Min/Max', () => <GridMinMax />)
+    .add('Custom Grid', () => <CustomGrid />)
+    .add('Partial Area Chart', () => <PartialAreaChart />)
+    .add('Partial Line Chart', () => <PartialLineChart />)
```
Same as above

LRNZ09 commented on storybook/stories/bar-stack/with-on-press.js:
```diff
@@ -0,0 +1,37 @@
+const keys = [ 'broccoli', 'celery', 'onions', 'tomato' ]
+const data = [ ...Array(times) ].map((_, index) =>
+    keys.reduce((obj, key) => ({ ...obj, [key]: getItem(index, key) }), {})
+)
```
Agree 👍

LRNZ09 commented:
Yeah sure! Actually I was thinking, why not put all the available `react-native-svg`'s [common props](https://github.com/react-native-community/react-native-svg#common-props) (including [touchable events](https://github.com/react-native-community/react-native-svg#touch-events) such as `onPress`) in a sub-key `svg` like how is done in `PieChart`?

JesperLekland commented:
That sounds perfectly reasonable! 👍  Basically that we always support passing down those props to each `react-native-svg` component

LRNZ09 commented:
Done! In README I've changed the examples language to `jsx` to enhance highlighting. I've also updated the `ShowcaseContainer` wrapping it in a `SafeAreaView` because without it trying the storybook on iPhone X cause the chart to go under the status bar.
